packages/wonder-blocks-timing/src/hooks/use-interval.js

```diff
@@ -0,0 +1,79 @@
+// @flow
+import {useEffect, useState, useCallback, useRef} from "react";
+
+import {
+    SchedulePolicy as SchedulePolicies,
+    ClearPolicy as ClearPolicies,
+} from "../util/policies.js";
+import type {IInterval, ClearPolicy, Options} from "../util/types.js";
+
+export function useInterval(
+    action: () => mixed,
+    intervalMs: number,
+    options?: Options,
+): IInterval {
+    if (typeof action !== "function") {
+        throw new Error("Action must be a function");
+    }
+
+    if (intervalMs < 1) {
+        throw new Error("Interval period must be >= 1");
+    }
+
+    const schedulePolicy =
+        options?.schedulePolicy ?? SchedulePolicies.Immediately;
+    const [isSet, setIsSet] = useState(
+        schedulePolicy === SchedulePolicies.Immediately,
+    );
+    const actionRef = useRef(action);
+    const mountedRef = useRef(false);
+
+    useEffect(() => {
+        mountedRef.current = true;
+        return () => {
+            mountedRef.current = false;
+        };
+    }, []);
```

Comment:
this would be nice as a hook by itself:
```js
const useMounted = () => {
    const ref  = useRef(false);
    useEffect(() => {
        ref.current = true;
        return () => ref.current = false;
    }, [])
    return ref
}
```

Reply:
Great idea.  We use this same pattern in `useTimeout`.  I'm going to keep it internal for now, but we can make it public later if we want.